symengine/pow.cpp

```diff
@@ -94,7 +91,14 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero)) {
+    if (is_a_Number(*b)) {
+        if(rcp_static_cast<const Number>(b)->is_negative())
+            throw std::runtime_error("pow: Zero cannot be raised to negative power");
+        }
```

Comment:
Due to [this](https://github.com/nishnik/symengine/blob/bug_fix_in_pow/symengine/pow.cpp#L90), `0**0` will evaluate to `1`. Should I move the condition `if(eq(*a, *zero))` above it, then only it will throw an error or should an extra condition be added `neq(*a, *zero)` in the first `if statement`?